src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -289,23 +291,31 @@ private static int CompareExpandedItem(CompletionItem item1, PatternMatch match1
             }
 
             // We prefer expanded item over non-expanded one iff the expanded item 
-            // is an exact match whereas the non-expanded one isn't.
+            // is an exact match whereas the non-expanded one is worse than prefix match.
             // 
-            // For example, suppose we have two types `Namespace1.Cafe1` and `Namespace2.Cafe`, and import completion is enabled.
-            // In the scenarios below, `Namespace2.Cafe` would be selected over `Namespace1.Cafe1`
-
+            // For example, suppose we have two items `Namespace1.Designer` and `Namespace2.DES`.
+            // In the scenarios below, `Namespace1.Designer` would be selected over `Namespace2.DES`.
+            //
             //  using Namespace1;
             //  class C
             //  {
-            //      cafe$$
+            //      des$$
+            //  }
+            //
+            // Or, if we have `Namespace1.MyTask` and `Namespace2.Task` in the example below, `Namespace2.Task` would be selected.
+            //
+            //  using Namespace1;
+            //  class C
+            //  {
+            //      task$$
             //  }
             if (isItem1Expanded && isItem1ExactMatch)
             {
-                return -1;
+                return match2.Kind != PatternMatchKind.Prefix ? -1 : 1;
             }
```

Comment:
let's chat

Reply:
i'm nto sure how this is actually correct.  what if item2 is an exact match here.  shouldn't it come first?

Reply:
you've combiend so many checks into so many conditions that i can't keep track of what states are possible, or what migth be missed.  we should really refactor all of this to be more clean.